tools/serverpod_cli/lib/src/database/extensions.dart

```diff
@@ -247,7 +250,7 @@ extension TableDefinitionPgSqlGeneration on TableDefinition {
     String out = '';
 
     // Table
-    out += 'CREATE TABLE "$name" (\n';
+    out += 'CREATE TABLE IF NOT EXISTS "$name" (\n';
```

Comment:
Suggestion: Remove "If not exists".

We hinted at this during our call but this comes with some migration implications that we should think about.

This will silence issues when we try and apply a migration to a database that does't not have the state we represent. Meaning a migration could completely fail to apply but go unnoticed and be updated in the migration registry. For example a table already exists but with the wrong columns.

We talked about this being OK when moving from an unmanaged state to a manged state (since that requires more knowledge from the user), but now this issue could occur for any database utalizing normal migrations.

Reply:
Alright, but this does add some complexity into the solution. Today we have no knowledge about this case in the migration / definition files. We will have to add a field for this in there to know if we should add the if not exists clause or not. 

I suppose this new variable also needs to be nullable or else we risk breaking backwards compatibility with the RC (perhaps ok?)